infra/gravity/node_commands.go

```diff
@@ -155,14 +155,79 @@ type ClusterStatus struct {
 	Cluster string `json:"domain"`
 	// State is the cluster state
 	State string `json:"state"`
-	// SystemStatus is the cluster status, see https://github.com/gravitational/satellite/blob/7.1.0/agent/proto/agentpb/agent.proto#L50-L54
-	SystemStatus int `json:"system_status"`
+	// SystemStatus is the cluster status, custom unmarshalling logic needed to accommodate Gravity differences
+	SystemStatus int
 	// Token is secure token which prevents rogue nodes from joining the cluster during installation
 	Token Token `json:"token"`
 	// Nodes describes the nodes in the cluster
 	Nodes []NodeStatus `json:"nodes"`
 }
 
+// unmarshalSystemStatus is a helper type to parse string or integer system statuses.
+//
+// For the original definitions and mappings see:
+//   https://github.com/gravitational/satellite/blob/5.0.2/agent/proto/agentpb/agent.proto#L36-L40
+//   https://github.com/gravitational/satellite/blob/5.0.2/agent/proto/agentpb/agent.pb.go#L48-L63
+func unmarshalSystemStatus(val json.RawMessage, status *int) error {
+	err := json.Unmarshal(val, &status)
+	if err != nil { // parsing as int failed, try parsing as a string and converting
+		var str string
+		if err2 := json.Unmarshal(val, &str); err2 != nil {
+			return trace.BadParameter("unable to parse system_status %q as int or string: %v, %v", val, err, err2)
+		}
+		switch str {
+		case "running":
+			*status = constants.SystemStatus_Running
+		case "degraded":
+			*status = constants.SystemStatus_Degraded
+		default:
+			return trace.BadParameter("unknown system_status: %q", str)
```

Comment:
Should this rather fallback to `*status = constants.SystemStatus_Unknown` or something like this instead of failing the entire marshaling?

Reply:
I spent some time exploring this.  Outcome-wise, it doesn't matter much either way.  Both implementations will fail after a 5 minute timeout.  One with the above BadParameter parse error and the other with the following error:

https://github.com/gravitational/robotest/blob/400f3eb5912f3647df1f29b027d89bedcf7cbbda/infra/gravity/cluster_status.go#L43

I think I prefer the BadParamerter parse error for two reasons:

1) converting string -> int is lossy.  The original string is more useful/indicative of the problem than the int `0` (or any int)
2) Keeping status parse errors distinct from "parseable but unhealthy" is useful for #181 -- one of my longer term goals for the code base.  Admittedly we don't make a distinction currently:

https://github.com/gravitational/robotest/blob/400f3eb5912f3647df1f29b027d89bedcf7cbbda/infra/gravity/cluster_status.go#L70-L83

But as part of this investigation I've got a patch in the works to make that distinction and fail fast on non-retryable failures during status collection.